crates/graphql/src/model/oauth.rs

```diff
@@ -108,17 +108,25 @@ impl OAuth2Session {
     }
 
     /// User authorized for this session.
-    pub async fn user(&self, ctx: &Context<'_>) -> Result<User, async_graphql::Error> {
+    pub async fn user(&self, ctx: &Context<'_>) -> Result<Option<User>, async_graphql::Error> {
         let state = ctx.state();
+        let Some(user_id) = self.0.user_id else {
+            return Ok(None);
+        };
+
+        if !ctx.requester().is_owner_or_admin(&UserId(user_id)) {
```

Comment:
This is just adding the same check as everywhere: you can get this property if you're either the owner of the property (which is only ever the case for a requester with a user in it) or if you're an admin (= you have the `urn:mas:admin` scope). It is not really necessary, because if you are allowed to access the OAuth2Session, you're allowed to access the underlying user, but it doesn't cost much to add the check here